src/payment/strategies/amazon-pay/amazon-pay-payment-strategy.spec.ts

```diff
@@ -470,26 +471,10 @@ describe('AmazonPayPaymentStrategy', () => {
     });
 
     describe('When 3ds is enabled', () => {
```

Comment:
I still can't see a test that verifies you're using the order reference from the `initializationData` as a fallback.